scanner/services/validators/validators.go

```diff
@@ -0,0 +1,132 @@
+package validators
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/quay/claircore/pkg/cpe"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+// hasIDAndCPE is the common interface offered by some proto definitions related
+// to ClairCore. It's used to simply validation.
+type hasIDAndCPE interface {
+	GetCpe() string
+	GetId() string
+}
+
+// ValidateGetVulnerabilitiesRequest validates matcher get vulnerability
+// requests, ensuring any potential internal error is captured upfront as invalid
+// argument.
+func ValidateGetVulnerabilitiesRequest(req *v4.GetVulnerabilitiesRequest) error {
+	if req == nil {
+		return errors.New("empty request")
```

Comment:
nit: the errors generated by the validators may benefit from of type `InvalidArgs` / `Bad Request`

Reply:
They are wrapped as invalid input by the caller: https://github.com/stackrox/stackrox/blob/c3d5fd9c3e24c4ebcfd660a5f4678ea0cff04e73/scanner/services/matcher.go#L47